js/event.ts

```diff
@@ -0,0 +1,242 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as domTypes from "./dom_types";
+
+const EVENT_ATTRIBUTES = new WeakMap;
+
+export enum EventPhase {
+  NONE = 0,
+  CAPTURING_PHASE = 1,
+  AT_TARGET = 2,
+  BUBBLING_PHASE = 3,
+}
+
+export class Event {
+  // Each event has the following associated flags
+  private stopPropagationFlag: boolean = false;
```

Comment:
> we should avoid the trivial notations of `: boolean`

Can you explain why on this piece? I don't see how this is a bad thing to declare the type of the property.

> I am also confused of the relationship between these flags and the EVENT_ATTRIBUTES weak map.

The event attributes are outlined in the spec interface and the flags are referred to in the definition. Since they aren't in the interface I didn't see a reason to expose them the same way.

Reply:
> Can you explain why on this piece? I don't see how this is a bad thing to declare the type of the property.

I could possibly see it with public properties, as it "might" make intent clearer, but if the property initialiser represents the value, that is just a valid as an annotation.  Someone looking at `private foo = false` can fully understand that intent of it as `private foo: boolean = false`, therefore it is _trivial_.  The tslint if configured to flag this in most situations.

> The event attributes are outlined in the spec interface and the flags are referred to in the definition. Since they aren't in the interface I didn't see a reason to expose them the same way.

I am still confused.  Why would you utilise two different mechanisms for private state?  I am also not sure the functional relationship between the two, they seem to overlap.  @kyranet and I debated and iterated on this before, if you are really concerned about private state and the downstream ramifications of it, while WeakMaps are likely how true private members will be polyfilled, a non-exported symbol of a module would suffice, like in `js/headers.ts`:

https://github.com/denoland/deno/blob/4e07783663d51877e7d41465cf5ef10d1540c4b3/js/headers.ts#L10-L14